response_operations_ui/views/collection_exercise.py

```diff
@@ -796,6 +816,77 @@ def get_event_name(tag):
     return event_names.get(tag)
 
 
+@collection_exercise_bp.route("/<short_name>/<period>/view-sample-ci", methods=["GET"])
+@login_required
+def get_view_sample_ci(short_name, period):
+    logger.info("Retrieving upload sample collection instrument page", short_name=short_name, period=period)
+    ce_details = build_collection_exercise_details(short_name, period)
```

Comment:
We seem to be building ce_details in build_collection_exercise_details and then updating it with attributes of itself, is then any reason it can't all go in build_collection_exercise_details?

Reply:
To be done in another card